sway-lib-std/src/vec.sw

```diff
@@ -3,11 +3,11 @@ library vec;
 use ::alloc::{alloc, realloc};
 use ::assert::assert;
 use ::intrinsics::size_of;
-use ::mem::{copy, read, write};
+use ::mem::{copy, ptr_offset, read, write};
 use ::option::Option;
 
 struct RawVec<T> {
-    ptr: u64,
+    ptr: raw_ptr,
```

Comment:
@digorithm what are the implications of this on the work the SDK team is doing to support `Vec`? A few things:
- `raw_ptr` is basically just a `u64` in the bytecode
- `raw_ptr` shows up as `raw untyped ptr` in the JSON ABI.
- It is not possible to initialize a `raw_ptr` without an explicit `asm` block.

Reply:
> `raw_ptr` is basically just a u64 in the bytecode.

Correct. I'm not sure if we'd want something different in the IR or bytecode since the point of this (as I understand it) is to prevent misuse of pointers earlier during compilation, ie. during type-checking.

> `raw_ptr` shows up as `raw untyped ptr` in the JSON ABI.

Is that a bad thing? I could make it show up as `raw_ptr` easily enough. Either way the SDK will have to know how to handle this type.

> It is not possible to initialize a `raw_ptr` without an explicit `asm` block.

Currently yes, though that's sort of intentional. Raw pointers are intended to be highly locked-down. You can obtain them via standard library functions like `alloc` and `stack_ptr` or by modifying an existing pointer using `ptr_offset`. Any other way that someone might want to create a pointer is likely to be highly unsafe.